File: SB/src/프로그래머스/레벨0/Day2/분수의_덧셈.java
package 프로그래머스.레벨0.Day2;

public class 분수의_덧셈 {
    class Solution {
        public int[] solution(int denum1, int num1, int denum2, int num2) {
            int num = num1 * num2; // 8
            int denum = (denum1 * num2) + (denum2 * num1); // 10
            int GCD = 1;
            for (int i = 1; i <= Math.min(num, denum); i++) {
                if (num % i == 0 && denum % i == 0) {
                    GCD = i;
                }
            }

            int[] answer = {denum / GCD, num / GCD};
            return answer;
        }
    }
}

File: HB/pysrc/프로그래머스/레벨0/Day2/분수의_덧셈.py
def solution(denum1, num1, denum2, num2):
    answer = []
    ud = num1*num2
    up = denum1*num2 + denum2*num1
    ran = ud
    for i in range(0,ran+1):
        if(ud % (ran-i) ==0 and up%(ran-i)==0):
            ud = ud//(ran-i)
            up = up//(ran-i)
            break
    answer.append(up)
    answer.append(ud)
            
    return answer